lib/realms/zones/in_play.rb

```diff
@@ -23,31 +22,35 @@ def actions
         base_actions + ally_actions + scrap_actions
       end
 
-      def automatic_actions
-        actions.select(&:auto?)
+      def execute
+        cards_in_play.each do |card|
+          perform(card.primary_ability) if card.auto_primary_available?
+          perform(card.ally_ability) if auto_ally_available?(card)
+        end
       end
 
       def cards_in_play
         in_play.values
       end
 
+      def reset!(card)
+        card_in_play = in_play.fetch(card.key)
+        card_in_play.reset!
+      end
+
       private
 
       def add_card(card)
         @in_play[card.key] = CardInPlay.new(card)
       end
 
       def base_actions
-        cards_in_play.select(&:base_activated?).map { |card| Actions::BaseAbility.new(active_turn, card) }
+        cards_in_play.reject(&:primary_used?).map { |card| Actions::BaseAbility.new(active_turn, card) }
       end
 
       def ally_actions
         cards_in_play.each_with_object([]) do |card, actions|
-          next unless card.ally_activated?
-
-          if in_play.except(card.key).values.any? { |cip| (cip.ally_factions & card.ally_factions).present? }
-            actions << Actions::AllyAbility.new(active_turn, card)
-          end
+          actions << Actions::AllyAbility.new(active_turn, card) if ally_available?(card)
         end
```

Comment:
Need to come up with a way such that instances of `CardInPlay` don't leak outside of the in-play zone.